oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/endpoint/DefaultOAuth2TokenRequestHeadersConverter.java

```diff
@@ -29,50 +28,54 @@
 import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
 
 /**
- * Utility methods used by the {@link Converter}'s that convert from an implementation of
- * an {@link AbstractOAuth2AuthorizationGrantRequest} to a {@link RequestEntity}
- * representation of an OAuth 2.0 Access Token Request for the specific Authorization
- * Grant.
+ * Default Converter used by the
+ * {@link OAuth2AuthorizationCodeGrantRequestEntityConverter} that convert from an
+ * implementation of an {@link AbstractOAuth2AuthorizationGrantRequest} to a
+ * {@link RequestEntity} representation of an OAuth 2.0 Access Token Request for the
+ * specific Authorization Grant.
  *
+ * @author Peter Eastham
  * @author Joe Grandja
- * @since 5.1
- * @see OAuth2AuthorizationCodeGrantRequestEntityConverter
+ * @since 6.3
  * @see OAuth2ClientCredentialsGrantRequestEntityConverter
  */
-final class OAuth2AuthorizationGrantRequestEntityUtils {
+public class DefaultOAuth2TokenRequestHeadersConverter<T extends AbstractOAuth2AuthorizationGrantRequest>
+		implements Converter<T, HttpHeaders> {
 
-	private static HttpHeaders DEFAULT_TOKEN_REQUEST_HEADERS = getDefaultTokenRequestHeaders();
+	private static final HttpHeaders DEFAULT_TOKEN_HEADERS = getDefaultTokenRequestHeaders();
```

Comment:
I've swapped it to also having a `setter` and no longer being `static final`.
This also meant I felt a lot better adjusting to using just `Accept:  application/json`, since if there is a real need to go back to `application/json;charset=UTF-8` or `application/x-www-form-urlencoded;charset=UTF-8` for consumers, they can configure that the same way they can configure the `encodeClientCredentialsIfRequired`

Reply:
I don't think I prefer to add `setDefaultHeaders(...)`. Instead, we can add fields:

* `private MediaType accept` (defaults to `MediaType.APPLICATION_JSON`)
* `private MediaType contentType` (defaults to `MediaType.APPLICATION_FORM_URLENCODED`)

and private methods:

* `setAccept(MediaType)`
* `setContentType(MediaType)`

These methods will only be called by the new package-private static factory method (see below conversation). The `static final` and `getDefaultTokenRequestHeaders()` can be removed.